primitives/account/src/basic_account.rs

```diff
@@ -209,6 +210,105 @@ impl AccountTransactionInteraction for BasicAccount {
 
         Ok(logs)
     }
+
+    fn commit_failed_transaction(
+        accounts_tree: &AccountsTrie,
+        db_txn: &mut WriteTransaction,
+        transaction: &Transaction,
+    ) -> Result<AccountInfo, AccountError> {
+        let key = KeyNibbles::from(&transaction.sender);
+
+        let account = accounts_tree
+            .get(db_txn, &key)
+            .or_else(|| {
+                if transaction.total_value() != Coin::ZERO {
+                    None
+                } else {
+                    Some(Account::Basic(BasicAccount {
+                        balance: Coin::ZERO,
+                    }))
```

Comment:
More out of interest: What would be a transaction leading to this code being executed?

Reply:
This code is inherited from _commit_outgoing_transaction_ and I think it is used for signaling transactions: i.e: if the transaction total value is 0 and the account does not exist, we just use a "mock" basic account.
It was added in commit _a5ed6b79d7ca2883fe19b28e8ca8da6aaafdb294_ and if I remember correctly there was some panic or some issue that we were experimenting, however I don't know exactly what a signaling transaction that fails means, so if that is never a possibility, we can just remove this